app/src/main/kotlin/com/wire/android/ui/calling/initiating/InitiatingCallViewModel.kt

```diff
@@ -116,27 +123,37 @@ class InitiatingCallViewModel @Inject constructor(
         )
     }
 
-    private suspend fun initiateCall() {
-        val conversationType = retrieveConversationTypeAsync().await()
-        startCall(
-            conversationId = conversationId,
-            conversationType = conversationType
-        )
-        callRinger.ring(R.raw.ringing_from_me)
+    internal suspend fun initiateCall() {
+        val conversationType = retrieveConversationType()
+        withContext(dispatchers.main()) {
+            val result = startCall(
+                conversationId = conversationId,
+                conversationType = conversationType
+            )
+            when (result) {
+                StartCallUseCase.Result.Success -> callRinger.ring(R.raw.ringing_from_me)
+                StartCallUseCase.Result.SyncFailure -> showNoConnectionDialog()
+            }
+        }
     }
 
-    fun navigateBack() = viewModelScope.launch {
+    private fun showNoConnectionDialog() {
+        initiatingCallState = initiatingCallState.copy(error = InitiatingCallError.NoConnection)
+    }
+
+    fun navigateBack() = viewModelScope.launch(dispatchers.main()) {
         wasCallHangUp = true
         navigationManager.navigateBack()
     }
 
-    fun hangUpCall() {
-        viewModelScope.launch {
-            launch { endCall(conversationId) }
-            launch {
-                navigateBack()
-                callRinger.stop()
-            }
-        }
+    fun hangUpCall() = viewModelScope.launch(dispatchers.io()) {
+        endCall(conversationId)
+        callRinger.stop()
+        navigateBack()
+    }
```

Comment:
Why did you remove all the coroutines ? We need to run them in parallel..
This way you will see some delay before the screen is closed. `endCall` will take some time before it's done

Reply:
you were launching 3 threads here, 1 containing another 2. Could we do it then with just 1 extra one to end the call and leave the current onging thread to run the tasks to stop the ring and navigate back?

Reply:
I am lunching 3 coroutines 1 parent and 2 children. This will not affect the app performance anymore since coroutines are cheap compared to threads. The most important thing here is that they should run in parallel